src/syntax/parser.ts

```diff
@@ -5,37 +5,51 @@ class SkippingLexer implements nearley.Lexer {
   lexer: moo.Lexer;
   skip: Set<string>;
 
-  constructor(lexer: moo.lexer) {
+  constructor(lexer: moo.Lexer) {
     this.lexer = lexer;
     this.skip = new Set(["WS", "COMMENT"]);
   }
 
   next() {
     do {
       const token = this.lexer.next();
-      if (!(token && this.skip.has(token.type))) {
+      if (!(token != null && this.skip.has(token.type!))) {
```

Comment:
why `!=`? 
It's better to use `!==`, but the original is cleaner IMO

Reply:
Also is `token.type!` valid syntax?

Reply:
This changes were made to the original code, because `tsconfig.json` throws error on this line. But it's fine now. I didn't go line by line during this merging, so I missed this line.